packages/babel-plugin-proposal-class-properties/test/fixtures/private/optional-chain-delete-property-with-transform/exec.js

```diff
@@ -0,0 +1,102 @@
+class Foo {
+  static #x = 1;
+  static #self = Foo;
+  static self = Foo;
+  static getSelf() { return this }
+
+  static test() {
+    const o = { Foo: Foo };
+    const deep = { very: { o } };
+    function fn() {
+      return o;
+    }
+    function fnDeep() {
+      return deep;
+    }
+    expect(() => delete Foo?.#self.unicorn).toThrow(TypeError);
```

Comment:
Since `unicorn` is non configurable, `delete Foo.#self.unicorn` will throw `TypeError`, so we are sure that the regular path is invoked indeed.

Reply:
clever!